Marlin/src/pins/stm32f1/pins_BTT_SKR_CR6.h

```diff
@@ -0,0 +1,184 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+#pragma once
+
+/**
+ * BigTreeTech SKR CR-6 (STM32F103RET6) board pin assignments
+ */
+
+#define DEFAULT_MACHINE_NAME "Creality3D"
+#define BOARD_INFO_NAME "BTT SKR CR-6"
+
+#if NOT_TARGET(__STM32F1__)
+  #error "Oops! Select an STM32F1 board in 'Tools > Board.'"
+#endif
+
+
+//
+// EEPROM
+//
+
+#if NO_EEPROM_SELECTED
+  #define I2C_EEPROM
+#endif
+
+/* I2C */
+#if ENABLED(I2C_EEPROM)
+  #define IIC_EEPROM_SDA       PB7
+  #define IIC_EEPROM_SCL       PB6
+
+  #define MARLIN_EEPROM_SIZE 0x1000                 // 4KB
+#elif ENABLED(SDCARD_EEPROM_EMULATION)
+  #define MARLIN_EEPROM_SIZE 0x1000                 // 4KB
+#endif
+
+#define E2END                (MARLIN_EEPROM_SIZE - 1) // 2KB
+
+
+//
+// Limit Switches
+//
+
+#define X_STOP_PIN          PC0
+#define Y_STOP_PIN          PC1
+#define Z_STOP_PIN          PC14
+
+#define FIL_RUNOUT_PIN      PC15
+
+//
+// Probe
+//
+#define PROBE_TARE_PIN      PA1
+#define PROBE_ACTIVATION_SWITCH_PIN   PC2   // Optoswitch to Enable Z Probe
+
+//
+// Steppers
+//
+#define X_ENABLE_PIN        PB14
+#define X_STEP_PIN          PB13
+#define X_DIR_PIN           PB12
+
+#define Y_ENABLE_PIN        PB11
+#define Y_STEP_PIN          PB10
+#define Y_DIR_PIN           PB2
+
+#define Z_ENABLE_PIN        PB1
+#define Z_STEP_PIN          PB0
+#define Z_DIR_PIN           PC5
+
+#define E0_ENABLE_PIN       PD2
+#define E0_STEP_PIN         PB3
+#define E0_DIR_PIN          PB4
+
+//
+// Release PB4 (Z_STEP_PIN) from JTAG NRST role
+//
+#define DISABLE_DEBUG
+
+//
+// Temperature Sensors
+//
+#define TEMP_0_PIN         PA0   // TH1
+#define TEMP_BED_PIN       PC3   // TB1
+
+//
+// Heaters / Fans
+//
+
+#define HEATER_0_PIN       PC8   // HEATER1
+#define HEATER_BED_PIN     PC9   // HOT BED
+
+#define FAN_PIN            PC6   // FAN
+#define FAN_SOFT_PWM
+
+#define CONTROLLER_FAN_PIN  PC7
+
+//
+// Display
+//
+
+#if ENABLED(CR10_STOCKDISPLAY)
+
+    #define BTN_ENC            PA15
+    #define BTN_EN1            PA9
+    #define BTN_EN2            PA10
+
+    #define LCD_PINS_RS        PB8
+    #define LCD_PINS_ENABLE    PB15
+    #define LCD_PINS_D4        PB9
+
+    #define BEEPER_PIN         PB5
+
+#endif
+
+//
+// USB connect control
+//
+#define USB_CONNECT_PIN    PA14
+#define USB_CONNECT_INVERTING false
+
+/**
+ * TMC2209 stepper drivers
+ * Hardware serial communication ports.
+ */
+#if HAS_TMC220x
+
+  //
+  // TMC2208 mode
+  //
+  // #define TMC2208_STANDALONE
+
+  #define X_HARDWARE_SERIAL  MSerial4
+  #define Y_HARDWARE_SERIAL  MSerial4
+  #define Z_HARDWARE_SERIAL  MSerial4
+  #define E0_HARDWARE_SERIAL MSerial4
+
+  //
+  // TMC2208 Software serial
+  //
+  // #define HAVE_SW_SERIAL
+
+  // Reduce baud rate to improve software serial reliability
+  // #define TMC_BAUD_RATE 19200
+
+#endif
```

Comment:
You should delete all this commented out code. I assume there is no 2208 version of this board?

You should also add the TMC slave addresses here. We have just started doing this for board with hard-wired addresses. Here is an example for the SKR E3 Mini V2 board:
https://github.com/MarlinFirmware/Marlin/blob/bugfix-2.0.x/Marlin/src/pins/stm32f1/pins_BTT_SKR_MINI_E3_V2_0.h

Reply:
Added the default slaves and removed the commented code.

Reply:
> You should also add the TMC slave addresses here. We have just started doing this for board with hard-wired addresses. Here is an example for the SKR E3 Mini V2 board:

Regarding this, what defines the place of such definitions? For instance, when using `CR10_STOCKDISPLAY`, you always need the serial ports to be defined a certain way for the BTT TFT to be able to connect.

Or even more: if you have this board, the first serial port is always -1, so why not define it in the pins file?

Reply:
Where did you find the addresses from? I can't find anything in the CR6 repo documenting it, and their firmware in that repo doesn't seem to actually be configured for this board?

I'm trying to look at a picture to see what the order is, but there is the minor issue that I don't know which chip is for which axis! Assuming they are arranged in a typical X/Y/Z/E0 order they don't _appear_ to be addressed sequentially. It's hard to tell for sure without a schematic.

![image](https://user-images.githubusercontent.com/20053467/102721869-4558ee00-42b2-11eb-8d99-bdfe441925fd.png)

Reply:
OK, I just chatted with you on GitHub. I see that the addresses actually are from their firmware on GitHub.

I'm not 100% sure they are correct, but it would take some physical probing and reversing on the board to confirm or deny these settings. For now we can trust their config 🙂

Reply:
Found a picture of the bottom of the board. The steppers in the picture above are in the order E/X/Y/Z, so the current order is probably correct.